doc/developer-notes.md

```diff
@@ -1080,6 +1080,10 @@ A few guidelines for introducing and reviewing new RPC interfaces:
     until the wallet is caught up to the chainstate as of the RPC call's entry.
     This also makes the API much easier for RPC clients to reason about.
 
+- The RPC call `getbalances` also returns `m_last_block_processed` in the `lastblock`
+  field. As this call locks the chainstate we can also inform about the block
+  the balances were valid on.
```

Comment:
Maybe worth mentioning the result shouldn't be cached because importing new keys could invalidate it. Instead of "were valid on" maybe say "were calculated/computed/evaluated on"?

Reply:
Thanks. Have updated the wording and added a warning regarding caching of results.